woocommerce/wc-admin
# Homescreen layout changes

Fixes #5223.

**This PR implements the following from #5223:**

- Default new stores to the single column homescreen
- Dismissing the task list, or completing the task list switches the homescreen to two columns
- Move task list / quick links to the left column
- Move inbox to right column under stats

**Additional behavior not explicitly described in the issue:**
- User preference for homescreen layout always takes precedent over the default (even if changed before the task list is complete)
- Left column (with the task list) is now the sticky column (used to be the right)
- The "your store today" header only displays when two columns are visible (gleaned from designs - not 100% if that's intended)

**Implementation notes**

The default layout (for new vs. existing stores) is controlled using a new option, `woocommerce_default_homepage_layout`. The client side code defaults to `single_column` if it's unset. Existing stores will have the value `two_columns` set in the version `1.7.0` DB update routine - while new stores will not. Again, any user preferences set will override this value.

### Screenshots

![Screen Shot 2020-10-22 at 1 41 28 PM](https://user-images.githubusercontent.com/63922/96909667-53ce8900-146c-11eb-8516-4d94ca4dfcc2.png)

### Detailed test instructions:

- On an existing store (be sure to clear any `woocommerce_admin_homepage_layout` user meta if you've tested https://github.com/woocommerce/woocommerce-admin/pull/5429):
  - Verify the home screen default gets set to `two_columns` (check the `woocommerce_default_homepage_layout` option after the DB updates run)
  - Verify that you can toggle to a single column view
- On a new store:
  - Verify that the home screen is single column by default
  - Dismiss the task list
  - Verify the home screen default switches to two columns
- On another new store:
  - Verify that the home screen is single column by default
  - Complete the task list
  - Verify the home screen default switches to two columns
- Verify the two column layout changes - task list on left, quick links on left, inbox under stats
- Verify single column layout changes (see issue) inbox on bottom



### Changelog Note:



Tweak: alter homescreen layout.
